monkeys/demanding-monkey/run.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -o pipefail
+
+ERROR_MSG="oh noez! demanding monkey got impatient."
+
+. $(dirname ${BASH_SOURCE})/helpers.bash
+
+init_monkey
+
+function log_line() {
+  MSG=$(/bin/date --rfc-3339=seconds)
+  MSG="${MSG}  $*"
+  echo "${MSG}"
+}
+export -f log_line
+
+# Values we depend on:
+#   DEBUG (option) - to spit out diagnostics
+#   DEBUG_EVENTS (optional) - write out event file to data
+
+if [ ${DEBUG} ]; then
+  set -x
+fi
+
+[ -z "$WATERMARK" ] && {
+        WATERMARK="10"
+}
+
+API_TOKEN_PATH=${API_TOKEN_PATH:-/var/run/secrets/kubernetes.io/serviceaccount/token}
+KUBE_TOKEN=$(< ${API_TOKEN_PATH})
+
+# Running namespace
+NAMESPACE_NAME=${NAMESPACE_NAME:-default}
+
+endpoint_debug_enable
+
+
+# access kubernetes and google.com - if WATERMARK exceeded, then will fail.
+function latency_test() {
+  while true; do
+    start_monitor
+    start_tcpdump
+    log_line "trying to get endpoints from Kubernetes"
+    curl -s -k -H "Authorization: Bearer $KUBE_TOKEN" -D /dev/stderr --fail --connect-timeout 3 --max-time $WATERMARK "https://kubernetes.default.svc.cluster.local/api/v1/namespaces/chaos-testing/endpoints" -w "time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}',Transfer '%{time_starttransfer}', total '%{time_total}'"
```

Comment:
Would be great to use a helper and potentially extract detailed timing statistics as described here:
https://blog.josephscott.org/2011/10/14/timing-details-with-curl/

In particular, DNS resolution can introduce a 5-10s delay in rare cases and we want to make sure that we can see at a first glance which portion of the connection was slow: DNS, connect, pre-transfer, transfer.

Reply:
TIL, this is awesome. Will update accordingly.

Reply:
Done.